### PR TITLE
[New Rule] AWS Suspicious SAML Activity

### LINKED ISSUE #1496: [New Rule] AWS Suspicious SAML Activity
## Description

Identifies when suspicious SAML activity has occurred in AWS.

### Platforms

Cloudtrail

### Query

```
event.dataset:aws.cloudtrail and event.provider:(iam.amazonaws.com or sts.amazonaws.com) and 
event.action:(Assumerolewithsaml or UpdateSAMLProvider) and event.outcome:success
```

### References
- https://docs.aws.amazon.com/IAM/latest/APIReference/API_UpdateSAMLProvider.html
- https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithSAML.html

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1496

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+[metadata]
+creation_date = "2021/09/22"
+maturity = "production"
+updated_date = "2021/09/22"
+integration = "aws"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies when suspicious SAML activity has occurred in AWS. An adversary could gain backdoor access via SAML manipulation.
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml:10]: If we don't differentiate between legit activity, we should remove the "suspicious" here

Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml:10]: Also let's replace "An adversary could gain backdoor access via SAML manipulation." with "An adversary could manipulate SAML to maintain access to the target."

File: rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    SAML Provider could be updated by a system administrator. Verify whether the user identity, user agent, and/or 
+    hostname should be making changes in your environment. SAML Provider being updated from unfamiliar users should 
+    be investigated. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-25m"
+index = ["filebeat-*", "logs-aws*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS Suspicious SAML Activity"
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml:23]: If we don't differentiate between legit activity, we should remove the "suspicious" here

File: rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+[[rule.threat.technique.subtechnique]]
+id = "T1484.002"
+name = "Domain Trust Modification"
+reference = "https://attack.mitre.org/techniques/T1484/002/"
+
+
+[rule.threat.tactic]
+id = "TA0004"
+name = "Privilege Escalation"
+reference = "https://attack.mitre.org/tactics/TA0004/"
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_aws_suspicious_saml_activity.toml:58]: I think https://attack.mitre.org/techniques/T1550/001/ and Defense Evasion would be a better fit here, we also need to rename the file

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**